src/data/mutable_data.rs

```diff
@@ -395,6 +395,42 @@ impl MutableData {
         Ok(())
     }
 
+    /// Mutates entries without performing any validation.
+    ///
+    /// For updates and deletes, the mutation is performed only if he entry version
+    /// of the action is higher than the current version of the entry.
+    pub fn mutate_entries_without_validation(&mut self, actions: BTreeMap<Vec<u8>, EntryAction>) {
```

Comment:
It would probably simplify `mutate_entries` if it would call `mutate_entries_without_validation` after the validation.
Or is `mutate_entries` not even needed by vaults anymore now, since validation always happens before the vote and mutation happens afterwards? (And the same goes for permissions and owners, I guess.)